pkg/stringutils/stringutils.go

```diff
@@ -32,6 +32,18 @@ func GenerateRandomASCIIString(n int) string {
 	return string(res)
 }
 
+// Ellipsis truncates a string to fit within maxlen, and appends ellipsis (...).
+// For maxlen of 3 and lower, no ellipsis is appended.
+func Ellipsis(s string, maxlen int) string {
+	if len(s) <= maxlen {
+		return s
+	}
+	if maxlen <= 3 {
+		return s[:maxlen]
+	}
+	return s[:maxlen-3] + "..."
```

Comment:
Also, you need to make sure you are not truncating in the middle of a UTF8 character...